packages/starknet-snap/src/state/network-state-manager.ts

```diff
@@ -0,0 +1,126 @@
+import { assert, string } from 'superstruct';
+
+import type { Network, SnapState } from '../types/snapState';
+import type { IFilter } from './filter';
+import { ChainIdFilter as BaseChainIdFilter } from './filter';
+import { StateManager, StateManagerError } from './state-manager';
+
+export type INetworkFilter = IFilter<Network>;
+
+export class ChainIdFilter
+  extends BaseChainIdFilter<Network>
+  implements INetworkFilter {}
+
+export class NetworkStateManager extends StateManager<Network> {
+  protected getCollection(state: SnapState): Network[] {
+    return state.networks;
+  }
+
+  protected updateEntity(dataInState: Network, data: Network): void {
+    assert(dataInState.name, string());
+
+    dataInState.name = data.name;
+    dataInState.nodeUrl = data.nodeUrl;
+  }
+
+  async addDefaultNetworks(networks: Network[]): Promise<void> {
```

Comment:
The name could be improved because it can do both update or add depending on if the network exists.

Reply:
it is not anupsert, but just add a set of default networks, thats why i add this name

let me add a document